tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -0,0 +1,518 @@
+	}, nil
+}
+
+// newSession uses AWS credentials to create and returns a session for interacting with EC2.
+func newSession(credentialPath, credentialAccountID, region string) (*awssession.Session, error) {
+	if _, err := os.Stat(credentialPath); os.IsNotExist(err) {
+		return nil, fmt.Errorf("failed to find AWS credentials from path '%v'", credentialPath)
+	}
+	session := awssession.Must(awssession.NewSession(&aws.Config{
+		Credentials: credentials.NewSharedCredentials(credentialPath, credentialAccountID),
+		Region:      aws.String(region),
+	}))
+	return session, nil
+}
+
+// CreateWindowsVM takes in imageId, instanceType, and sshKey name to create a Windows instance under the same VPC as
+// the existing OpenShift cluster with the following:
+// - attaches existing cloud-specific cluster worker security group and IAM to gain the same access as the linux
+// workers,
+// - uses public subnet,
+// - attaches public ip to allow external access,
+// - adds a security group that allows traffic from within the VPC range and RDP access from user's IP,
+// - uses given image id, instance type, and sshKey name
+// - creates a unique name tag for the instance using the same prefix as the OpenShift cluster name, and
+// - logs id and security group information of the created instance in 'windows-node-installer.json' file at the
+// resourceTrackerDir.
+// On success, the function outputs RDP access information in the commandline interface.
+func (a *awsProvider) CreateWindowsVM(imageId, instanceType, sshKey string) error {
+	infraID, err := a.openShiftClient.GetInfrastructureID()
+	if err != nil {
+		return err
+	}
+	vpcID, err := a.getVPCIdByInfrastructure(infraID)
+	if err != nil {
+		return fmt.Errorf("failed to find cluster VPC, %v", err)
+	}
+	err = a.ensureDnsOnVpcEnabled(vpcID)
+	if err != nil {
+		log.Info("failed to ensure DNS is enabled on VPC, %v", err)
+	}
+	workerSG, err := a.getClusterWorkerSGID(infraID)
+	if err != nil {
+		log.Info("failed to cluster worker security group, please attach manually")
+	}
+	workerIAM, err := a.getIAMWorkerRole(infraID)
+	if err != nil {
+		log.Info("failed to cluster worker IAM role, please attach manually")
+	}
+	publicSubnetID, err := a.getPublicSubnetId(vpcID, infraID)
+	if err != nil {
+		log.Info("failed to grab a Public subnet, please attach manually")
+	}
+	sgID, err := a.findOrCreateSg(infraID, vpcID)
+	if err != nil {
+		log.Info("failed to find or create security group under VPC, please create manually")
+	} else {
+		err = a.authorizeSgIngress(sgID)
+		if err != nil {
+			log.Info("failed to attach sg: %s, please attach manually", sgID)
+		}
+	}
+	instanceID, err := a.createInstances(imageId, instanceType, sshKey, publicSubnetID, sgID, workerSG, workerIAM)
+	if err != nil {
+		return err
+	}
+	// Associate unique tags and IP address to all created instances
+	_, err = a.createInstanceNameTag(instanceID, infraID)
+	if err != nil {
+		log.Info("failed to assign name for instance: %s", instanceID)
```

Comment:
This is a case where a `log.Warn` is sufficient and you don't need to return an error. Please replace the `log.Info` with `log.V(0) ` given there is no `log.Warn()`. But please check if it is available.